packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -48,19 +52,46 @@ export async function createNewFluidFile(
     logger: ITelemetryLogger,
     createNewSummary: ISummaryTree | undefined,
     epochTracker: EpochTracker,
+    fileEntry: IFileEntry,
+    createNewCaching: boolean,
 ): Promise<IOdspResolvedUrl> {
     // Check for valid filename before the request to create file is actually made.
     if (isInvalidFileName(newFileInfo.filename)) {
         throwOdspNetworkError("Invalid filename. Please try again.", invalidFileNameStatusCode);
     }
 
-    const itemId = createNewSummary === undefined
-        ? await createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker)
-        : await createNewFluidFileFromSummary(getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+    let itemId: string;
+    let id: string;
+
+    if (createNewSummary === undefined) {
+        itemId = await createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker);
+        id = "";
```

Comment:
You can remove this and assert below. See my other comment.

Reply:
if I do that , the assert will get an error , id is used before it's assigned, I can move this to declaration